libraries/ESP8266WiFi/src/include/ClientContext.h

```diff
@@ -456,8 +458,10 @@ class ClientContext
             }
 
             _send_waiting = true;
-            // This delay will be interrupted by esp_schedule on next received ack
-            delay(_timeout_ms);
+            for (decltype(_timeout_ms) i = 0; _send_waiting && i < _timeout_ms; i++) {
+               // let a chance to recurrent scheduled functions (ethernet)
```

Comment:
I would also keep the old comments because delays can and will still be interrupted by `esp_schedule` and it needs to be clear in this code.
```
// This delay will be interrupted by esp_schedule in the connect callback
// This delay will be interrupted by esp_schedule on next received ack
```